python/ray/autoscaler/_private/vsphere/node_provider.py

```diff
@@ -187,19 +186,14 @@ def node_tags(self, node_id):
             return dict(d1)
 
     def external_ip(self, node_id):
-        # Return the external IP of the VM
-
-        vm = self.vsphere_sdk_client.vcenter.vm.guest.Identity.get(node_id)
-        try:
-            _ = ipaddress.IPv4Address(vm.ip_address)
-            logger.debug("Fetch IP {} for VM {}".format(vm.ip_address, vm))
-        except ipaddress.AddressValueError:
-            # vSphere SDK could return IPv6 address when the VM is just booted. We
-            # just return None in this case because the Ray doesn't support IPv6
-            # address yet When the next time external_ip is called, we could return
-            # the IPv4 address
-            return None
-        return vm.ip_address
+        vm = self.pyvmomi_sdk_provider.get_pyvmomi_obj_by_moid(
+            [vim.VirtualMachine], node_id
+        )
+        for ipaddr in vm.guest.net[0].ipAddress:
```

Comment:
Could you add some comments explain the thoughts?

Reply:
Done.